system/daemon.go

```diff
@@ -0,0 +1,526 @@
+	var u *Unit
+	if u, err = sys.Get(name); err != nil {
+		return
+	}
+
+	if reloader, ok := u.Interface.(unit.Reloader); ok {
+		return reloader.Reload()
+	}
+
+	return ErrNoReload
+}
+
+// TODO
+func (sys *Daemon) Enable(name string) (err error) {
+	var u *Unit
+	if u, err = sys.Get(name); err != nil {
+		return
+	}
+	u.Log.Println("enable")
+	return ErrNotImplemented
+}
+
+// TODO
+func (sys *Daemon) Disable(name string) (err error) {
+	var u *Unit
+	if u, err = sys.Get(name); err != nil {
+		return
+	}
+	u.Log.Println("disable")
+	return ErrNotImplemented
+}
+
+// IsEnabled returns enable state of the unit held in-memory under specified name
+// If error is returned, it is going to be ErrNotFound
+func (sys *Daemon) IsEnabled(name string) (st unit.Enable, err error) {
+	//var u *Unit
+	//if u, err = sys.Unit(name); err == nil && sys.Enabled[u] {
+	//st = unit.Enabled
+	//}
+	return unit.Enabled, ErrNotImplemented
+}
+
+// IsActive returns activation state of the unit held in-memory under specified name
+// If error is returned, it is going to be ErrNotFound
+func (sys *Daemon) IsActive(name string) (st unit.Activation, err error) {
+	var u *Unit
+	if u, err = sys.Get(name); err == nil {
+		st = u.Active()
+	}
+	return
+}
+
+var std = New()
+
+// Get looks up the unit name in the internal hasmap of loaded units and calls
+// sys.Load(name) if it can not be found
+// If error is returned, it will be error from sys.Load(name)
+func (sys *Daemon) Get(name string) (u *Unit, err error) {
+	var ok bool
+	if u, ok = sys.loaded[name]; !ok {
+		u, err = sys.Load(name)
+	}
+	return
+}
+
+// StatusOf returns status of the unit held in-memory under specified name
+// If error is returned, it is going to be ErrNotFound
+func (sys *Daemon) StatusOf(name string) (st unit.Status, err error) {
+	var u *Unit
+	if u, err = sys.Get(name); err != nil {
+		return
+	}
+
+	st = unit.Status{
+		Load: unit.LoadStatus{
+			Path:   u.Path(),
+			Loaded: u.Loaded(),
+			State:  unit.Enabled,
+		},
+		Activation: unit.ActivationStatus{
+			State: u.Active(),
+			Sub:   u.Sub(),
+		},
+	}
+
+	st.Log, err = ioutil.ReadAll(u.Log)
+
+	return
+}
+
+// Load searches for a definition of unit name in configured paths parses it and returns a pointer to Unit
+// If a unit name has already been parsed(tried to load) by sys, it will not create a new unit, but return a pointer to that unit instead
+func (sys *Daemon) Load(name string) (u *Unit, err error) {
+	if !Supported(name) {
+		return nil, ErrUnknownType
+	}
+
+	var paths []string
+	if filepath.IsAbs(name) {
+		paths = []string{name}
+	} else {
+		paths = make([]string, len(sys.Paths))
+		for i, path := range sys.Paths {
+			paths[i] = filepath.Join(path, name)
+		}
+	}
+
+	for _, path := range paths {
+		var file *os.File
+		if file, err = os.Open(path); err != nil {
+			if os.IsNotExist(err) {
+				continue
+			}
+			return nil, err
+		}
+		defer file.Close()
+
+		var parsed bool
+		if u, parsed = sys.parsed[name]; !parsed {
```

Comment:
yea - so what I'm suggesting is that you check for the unit, and if it is in `loaded`, return it immediately instead of leaving that check till later.  it should also get rid of a layer of indirection.